src/main/java/com/civclassic/oldenchanting/OldEnchanting.java

```diff
@@ -302,13 +306,41 @@ private void createXPBottles(Player player, int totalExp) {
 	
 	@EventHandler(priority = EventPriority.HIGHEST)
 	public void xpBottleEvent(ExpBottleEvent event) {
-		if(noExp) {
-			ProjectileSource source = event.getEntity().getShooter();
-			if(source instanceof Player) {
+		if (noExp) {
+			ThrownExpBottle bottle = event.getEntity();
+			ProjectileSource source = bottle.getShooter();
+			if (source instanceof Player) {
 				Player shooter = (Player) source;
 				shooter.giveExp(xpPerBottle);
+				bottle.teleport(shooter.getEyeLocation());
```

Comment:
Why is this needed

Reply:
When players throw Bottles o' Enchanting, they receive the xp on impact, regardless of the impact's location relative to the player. It means that the orbs themselves play no part in the player's leveling up, so the orbs will linger at the impact site until they expire or picked up.

This change teleports the orbs to the player at the moment of impact and in doing so removing them from the game but also providing the player with an audible feedback for receiving the xp.